### PR TITLE
Enh: Oauth2

### PR DESCRIPTION
**Pull request by Felli**
Switch from Oauth1 to Oauth2 use.

> Note: This also supports Font Awesome 5 icons for those that use them.

### CODE REVIEW

File: protected/humhub/modules/user/authclient/Twitter.php [hunk 1]
```diff
@@ -20,49 +21,32 @@ protected function defaultViewOptions()
         return [
             'popupWidth' => 860,
             'popupHeight' => 480,
-            'cssIcon' => 'fa fa-twitter',
+            'cssIcon' => 'fab fa-twitter',
```
Comment (githubjeka) [inline protected/humhub/modules/user/authclient/Twitter.php]: fab ?

Comment (Felli) [reply protected/humhub/modules/user/authclient/Twitter.php]: `fab` is used in Font Awesome 5 `fa` is deprecated.

Comment (githubjeka) [reply protected/humhub/modules/user/authclient/Twitter.php]: Ok

Comment (githubjeka) [reply protected/humhub/modules/user/authclient/Twitter.php]: But https://github.com/humhub/humhub/blob/958d940da60080d1a2cf2f02b429826c865b5a79/composer.json#L40

version for HumHub is `>=4.7.0 <5.0.0`

Comment (Felli) [reply protected/humhub/modules/user/authclient/Twitter.php]: Sadly using `fa fa-twitter` or even `fa-twitter` wouldn't work anyways (Which are 4+), would also seem that they still have this issue within 5...

Comment (luke-) [reply protected/humhub/modules/user/authclient/Twitter.php]: @Felli Maybe we can change to FA 5.0 in the v1.3?

**Comment by luke-** [comment]
@Felli can you please recreate this PR directly for v1.3?

**Pull request closed without merging**